app/modules.go

```diff
@@ -156,3 +153,13 @@ func GenTxValidator(msgs []sdk.Msg) error {
 	}
 	return nil
 }
+
+func assertMsgType[T sdk.Msg](m sdk.Msg, idx int) error {
```

Comment:
Nice, I didn't know it could do this kind of function :tada:

Reply:
Generics :) Although it's not intuitive to create a zero type of type `T`.... see line 159